<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="omav">

    <xacro:arg name="resources_prefix" default=""/>
    <xacro:arg name="resources_directory" default="$(find mppi_omav_raisim)"/>

    <xacro:property name="resources_prefix" value="$(arg resources_prefix)"/>
    <xacro:property name="resources_directory" value="$(arg resources_directory)"/>
    <xacro:property name="mass_ouzel" value="5.0"/>
    <xacro:property name="ixx_ouzel" value="0.12"/>
    <xacro:property name="iyy_ouzel" value="0.14"/>
    <xacro:property name="izz_ouzel" value="0.18"/>

    <material name="green">
            <color rgba="0 1 0 1"/>
    </material>

    <link name="odom"/>

    <joint name="omav_pos_x" type="prismatic">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="odom"/>
        <child link="omav_x"/>
        <axis xyz="1 0 0"/>
        <limit effort="100" velocity="3.0"/>
    </joint>

    <link name="omav_x">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="omav_pos_y" type="prismatic">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="omav_x"/>
        <child link="omav_y"/>
        <axis xyz="0 1 0"/>
        <limit effort="100" velocity="3.0"/>
    </joint>

    <link name="omav_y">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="omav_pos_z" type="prismatic">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="omav_y"/>
        <child link="omav_z"/>
        <axis xyz="0 0 1"/>
        <limit effort="100" velocity="3.0"/>
    </joint>

    <link name="omav_z">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="omav_rot_x" type="continuous">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="omav_z"/>
        <child link="omav_rotation_x"/>
        <axis xyz="1 0 0"/>
        <limit effort="100"  velocity="1.5"/>
    </joint>

    <link name="omav_rotation_x">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="omav_rot_y" type="continuous">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="omav_rotation_x"/>
        <child link="omav_rotation_y"/>
        <axis xyz="0 1 0"/>
        <limit effort="100"  velocity="1.5"/>
    </joint>

    <link name="omav_rotation_y">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
    </link>

    <joint name="omav_rot_z" type="continuous">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="omav_rotation_y"/>
        <child link="omav_rotation_z"/>
        <axis xyz="0 0 1"/>
        <limit effort="100"  velocity="1.5"/>
    </joint>

    <link name="omav_rotation_z">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="${mass_ouzel}"/>
            <inertia ixx="${ixx_ouzel}" ixy="0.0" ixz="0.0" iyy="${iyy_ouzel}" iyz="0.0" izz="${izz_ouzel}"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://mppi_omav_raisim/data/meshes/ouzel_inspection_full.dae"/>
            </geometry>
        </visual>
    </link>

    <joint name="fixed" type="fixed">
        <parent link="odom"/>
        <child link="goal"/>
        <origin xyz="5.0 5.0 5.0" rpy="0 0 0"/>
    </joint>

    <link name="goal">
        <inertial>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <mass value="1.0"/>
            <inertia ixx="0.001667" ixy="0.0" ixz="0.0" iyy="0.0841667" iyz="0.0" izz="0.084167"/>
        </inertial>
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
            <material name="green"/>
            <geometry>
                <box size="0.2 0.2 0.2" />
            </geometry>
        </visual>
    </link>

</robot>
